.github/workflows/workflow.yml

```diff
@@ -0,0 +1,53 @@
+name: CI workflow for rebel hub
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    # uwuntu latest for simplicity
+    # python 3.12 container to match our proj
+    name: Build & Run Tests
+    runs-on: ubuntu-latest
+    container: python:3.12-slim
+    steps:
+      # fist we checkout the repo
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      # we install the backend deps listed in requirements.txt
+      - name: Install Backend Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r backend/requirements.txt
+
+      # HERE WE CAN ADD IN STEPS FOR LINTING/SECURITY/AND FORMATTING
```

Comment:
Will we decide on these steps in a meeting, or should we just discuss it on discord?